internal/tiltfile/tiltfile_state.go

```diff
@@ -1479,6 +1483,35 @@ func (s *tiltfileState) translateLocal() ([]model.Manifest, error) {
 	return result, nil
 }
 
+//this deals with alerting the user if there are multiple instances of a
+//the same yaml entity (same name:kind:namespace:group) which may have been
+//defined in the same way or in multiple ways.
+func (s *tiltfileState) warnDuplicateYamlEntities(m model.Manifest) {
+	deployTarget := m.K8sTarget()
+	displayNames := deployTarget.DisplayNames
+
+	duplicates := make(map[string]int)
+	//when the same entity appears more than once, Tilt handles it by storing it as name:kind:namespace:group:n where n is
+	//the nth repeat of that entity, so as to not cause execution to halt. In the future, it might be better to surface an
+	//error about the duplicated resource closer to the actual manifest assembly.
+	for _, name := range displayNames {
+		tempName := strings.Split(name, ":")
+		if len(tempName) > 4 { //this means there's an n which was appended at the manifest assembly because of multiple same entities present
+			duplicates[strings.Join(tempName[:4], ":")]++
+		}
+	}
+
+	//get the names of all the duplicates
+	var duplicateNames []string
+	for key := range duplicates {
+		duplicateNames = append(duplicateNames, key)
+	}
+	if len(duplicateNames) != 0 {
+		s.logger.Warnf("Resource %s contains multiple specifications of k8s entity(s): %s. Only one can be applied to the cluster; to ensure expected behavior, remove the duplicate specifications",
```

Comment:
Whoops I realize the error message reads a little funny when there are multiple duplicates detected. wdyt of something like this?

```suggestion
		s.logger.Warnf("Resource %s contains multiple specifications of k8s entity(s): %s. Only one specification per entity can be applied to the cluster; to ensure expected behavior, remove the duplicate specifications",
```

Reply:
Oh yeah thats fair, so if there's doggos and cats, it'll seem like you can have either only doggos or only cats. I'll fix that!